src/bands.cpp

```diff
@@ -93,7 +93,7 @@ void fields::prepare_for_bands(const vec &p, double endtime, double fmax,
 
   double cutoff_freq = 0.0;
   if (gv.dim == Dcyl) {
-    cutoff_freq = 1.84*a*dt/(2*pi)/gv.nr()/sqrt(epsmax);
+    cutoff_freq = 1.84*a*dt/(2*pi)/static_cast<double>(gv.nr())/sqrt(epsmax);
```

Comment:
Those casts were inserted to silence compiler warnings. Converting from  a 64 bit int to a double is a narrowing operation (the double mantissa has only 52 bits). To find all potential problems with the new 64 bit int type I had to use the gcc option `-Wconversion` that warns when an implicit conversion may alter the value. The meep::integer -> double conversions gave hundreds of false positives which made it hard to find the real problems. So I decided to insert static casts to make the conversion explicit. It turns of the compiler warning and documents that the conversion is actually intented. It looks a bit ugly but it does not cause any overhead (the conversion has to happen anyway, no matter if it is implicit or explicit). So I would prefer to leave the casts.